src/RouterProvider.tsx

```diff
@@ -9,50 +9,48 @@ import type { ExtendedOptions, PageObject } from './commonTypes';
 export default function RouterProvider({
   pageObject,
   options,
-  children,
+  children: initialChildren,
+  makePage,
 }: {
   pageObject: PageObject;
   options: ExtendedOptions;
   children: JSX.Element;
+  makePage: (route: string) => Promise<JSX.Element>;
 }) {
-  const [routerMock, setRouterMock] = useState<NextRouter>();
   const isMounted = useMountedState();
 
   const pushHandler = useCallback(async (url: Parameters<PushHandler>[0]) => {
     const nextRoute = url.toString();
-    const nextOptions = {
-      ...options,
-      route: nextRoute,
-    };
+    const nextOptions = { ...options, route: nextRoute };
+
     const nextPageObject = await getPageObject({
```

Comment:
Since `makePage` in `src/getPage` returns `{ pageElement, pageData, pageObject }`, can't we directly get `pageObject` from it (line 29)?